teamwork/apps/projects/views/EditProjectView.py

```diff
@@ -38,165 +37,43 @@ def edit_project(request, slug):
-                        profAdded = True
-                else:
-                    # add user to pending invitations
-                    project.pending_invitations.add(mem_to_add)
-                    project.save()
-
-                    # send user an alert
-                    Alert.objects.create(
-                        sender=request.user,
-                        to=mem_to_add,
-                        msg="You have been invited to join the Project: " + project.title,
-                        url=reverse('view_one_project',args=[project.slug]),
-                        alertType="invitation",
-                        slug=project.slug
-                        )
-
-                    # send user an email
-                    subject = "GrepThink Project Invitation: " + project.title
-                    content = "You have been invited to Join the Project: {0},\n\n You can accept this invitation from the alerts dropdown in the topright @ grepthink.com".format(project.title)
-
-                    send_email(mem_to_add, request.user.email, subject, content)
-                    added = True
-
-        if added:
-            messages.add_message(request, messages.SUCCESS, "Greppers have been invited to join your project!")
-        elif profAdded:
-            messages.add_message(request, messages.SUCCESS, "Greppers have been added to the project.")
-        else:
-            messages.add_message(request, messages.WARNING, "Failed to invite member(s) to project. Make sure they are enrolled in this course.")
-
-        return redirect(view_one_project, project.slug)
+        add_member_to_project(request, project)
 
     # Remove a user from the project
     if request.POST.get('remove_user'):
-        f_username = request.POST.get('remove_user')
-        f_user = User.objects.get(username=f_username)
-        to_delete = Membership.objects.filter(user=f_user, project=project)
-
-        remaining = Membership.objects.filter(project=project).exclude(user=f_user)
-
-        # check if they were the only member of the project
-        if len(members) == 1:
-            messages.warning(request,
-             "As the only member of the project, you must invite another to be the Project Owner, or delete the project via Edit Project!")
-        else:
-            # check if user that is being removed was Project Owner
-            if f_user == project.creator:
-                project.creator = remaining.first().user
-            # check if user that is being removed was Scrum Master
-            if f_user == project.scrum_master:
-                project.scrum_master = remaining.first().user
-
-            project.save()
-            messages.info(request, "{0} has been removed from the project".format(f_username))
-
-            # delete membership
-            for mem_obj in to_delete:
-                mem_obj.delete()
-
-        return redirect(view_one_project, project.slug)
-
+        remove_user_from_project(request, project)
```

Comment:
Im rusty here, do we no longer need to (or did we ever really need to) return a redirect from these functions?

Reply:
There is now only one render at the end. Before if you removed a user from the project etc, it would re-render the page immediately, discarding any other data that might have changed in that initial post request.